### PR TITLE
docs: Improve upgrade instructions

### PR DESCRIPTION
**Pull request by tara-det-ai**
Improve upgrade instructions per feedback.

**Updates to Upgrade and Installation Instructions**

Users may find the upgrade process onerous if they have not saved the commands (and any flags) they used when they installed Determined. Feedback on the upgrade instructions indicates the users may not realize they'll need to repeat the same steps they followed when they first installed Determined.

This PR resolves these issues.

**Out of Scope**

Provide specific upgrade instructions for each individual type of installation. See: TECHWR-314

https://hpe-aiatscale.atlassian.net/jira/software/projects/TECHWR/boards/41?selectedIssue=TECHWR-314

**Jira**

TECHWR-318

**Additional Updates**

In general, the on-prem installation navigation section is onerous for users because there are many options. It is not clear which page is an on-prem option and which page is simply requirements.

In the architecture introduction, the section "Interactive Job Configuration" is misleading. The content is about how to configure interactive jobs and the behavior of interactive jobs.

In the introduction, the section on commands and shells is outdated. This section was written before Core API was available. Now, a developer can use the `det cmd` Command to run a command when not running a trial, else they use an experiment.

The Elastic Infrastructure section should be updated to remove "deep learning" from the description. We should call out that the calculation made by the master when deciding the ideal size of the cluster takes into account the total resource requests of all workloads.

### CODE REVIEW

**Review by lbliii — APPROVED** [review]

File: docs/architecture/introduction.rst [hunk 1]
```diff
@@ -60,12 +60,11 @@ of the tasks you can perform with the Determined CLI:
...
 
-In addition to structured model training workloads, which are handled using :ref:`experiments
-<experiments>`, Determined also supports more free-form tasks using :ref:`commands and shells
-<commands-and-shells>`.
+:ref:`Commands and shells <commands-and-shells>` support free-form tasks.
 
-Commands execute a user-specified program on the cluster. Shells start SSH servers that allow using
-cluster resources interactively.
+In Determined, a developer uses an :ref:`experiment <experiments>`, to run a trial. When not running
+a trial, a developer can use a ``det cmd`` Command. Commands execute a user-specified program on the
```
Comment (ioga) [inline docs/architecture/introduction.rst:66]: should this be a lowercase command?
```suggestion
a trial, a developer can use a ``det cmd`` command. Commands execute a user-specified program on the
```

File: docs/architecture/introduction.rst [hunk 2]
```diff
@@ -250,9 +249,10 @@ following benefits:
 Elastic Infrastructure
 ======================
 
-When running in a cloud environment, Determined can automatically provision and terminate GPU
-instances as the set of deep learning workloads on the cluster changes. This capability is called
-*elastic infrastructure*. The agents that are provisioned by the system are called *dynamic agents*.
+When running in an AWS or GCP cloud environment, Determined can automatically provision and
+terminate GPU instances as the set of deep learning workloads on the cluster changes. This
```
Comment (ioga) [inline docs/architecture/introduction.rst:253]: should we strike out `deep learning`? user can run any workload. 

```suggestion
terminate GPU instances as the set of workloads on the cluster changes. This
```

File: docs/architecture/introduction.rst [hunk 3]
```diff
@@ -265,8 +265,9 @@ Following the diagram, the execution would be:
    workloads (agents waiting to be scheduled).
 
 #. The master calculates the ideal size of the cluster and decides how many agents to launch and
-   which agents to terminate. The calculation is done based on the configured scaling behavior and
-   the specification of the resource pools.
+   which agents to terminate. The calculation is based on the configured scaling behavior (minimum
+   and maximum amount of instances the master can spawn) and the specification of the resource
+   pools.
```
Comment (ioga) [inline docs/architecture/introduction.rst:270]: should we call out this calculation takes into account the total resource requests of all workloads? I am not sure it's implied or obvious.

>The calculation is based on the total resource requests of all jobs submitted to the cluster, configured scaling behavior (minimum and maximum amount of instances the master can spawn) and the specification of the resource pools.

**Review by ioga — APPROVED** [review]

File: docs/architecture/introduction.rst (hunk 3 above)
Comment (tara-det-ai) [reply docs/architecture/introduction.rst:272]: resolved

**Review by rb-determined-ai — APPROVED** [review]
codeowner stamp

**Pull request merged**